Sources/Sentry/SentryClient.m

```diff
@@ -49,8 +50,11 @@ - (_Nullable instancetype)initWithOptions:(SentryOptions *)options
                                             andMachineContextWrapper:machineContextWrapper];
 
         NSError *error = nil;
-        self.fileManager = [[SentryFileManager alloc] initWithDsn:self.options.parsedDsn
-                                                 didFailWithError:&error];
+
+        self.fileManager =
+            [[SentryFileManager alloc] initWithDsn:self.options.parsedDsn
+                               currentDateProvider:[[SentryDefaultCurrentDateProvider alloc] init]
+                                  didFailWithError:&error];
```

Comment:
M: what's the chance that the `SentryDefaultCurrentDateProvider` will fail and `error` will be not null? because then `SentryFileManager` will take a nullable `SentryDefaultCurrentDateProvider` and this could cause problems, should we null check the `SentryDefaultCurrentDateProvider` first and only then init. the `SentryFileManager`?

Reply:
`[SentryDefaultCurrentDateProvider alloc] init]` is a default constructor and should never fail. We handle the error of the init of the `SentryFileManager` two lines below. Did you maybe think the `error` was related to `SentryDefaultCurrentDateProvider`?

Reply:
mmmm got it, yes, maybe the indentation either from the code or GH messed up the indentation and this made me think that the error comes from the `SentryDefaultCurrentDateProvider`, my bad then

Reply:
<img width="740" alt="Screenshot 2020-07-28 at 12 02 51" src="https://user-images.githubusercontent.com/5731772/88651758-4eaaa980-d0ca-11ea-95e5-8bbd99544d9d.png">

Reply:
Sem problemas ✌️